event-publisher-spi/src/main/scala/uk/gov/nationalarchives/eventpublisherspi/EventPublisherProviderFactory.scala

```diff
@@ -21,7 +23,16 @@ class EventPublisherProviderFactory extends EventListenerProviderFactory {
     eventPublisherConfig = Option(EventPublisherConfig(snsUrl, snsTopicArn, tdrEnvironment))
   }
 
-  override def postInit(factory: KeycloakSessionFactory): Unit = { }
+  override def postInit(factory: KeycloakSessionFactory): Unit = {
+    factory.register(event => {
+      if(event.isInstanceOf[PostMigrationEvent]) {
+         val session = factory.create()
+        val provider: TimerProvider = session.getProvider(classOf[TimerProvider])
+        val task = UserMonitoringTask(eventPublisherConfig.get)
```

Comment:
Cold the task val be renamed: userMonitoringTask just to make it clear what task it is? Just thinking about when we come to add more tasks to run.

Reply:
I've renamed it to userMonitoringTask